### ISSUE TITLE
ComboBox - Accessibility - Pointer Cancellation Violation

### ISSUE THREAD

**Issue by malanirishi** [adobe/react-spectrum#2141]
# 🐛 Bug Report 

Pointer Cancellation is accessibility functionality whereby users with limited motor skills (and frequently mis-click) are able to "cancel" their clicks before they change the state of the app.  In our "Move" dialog, the comboboxes violate this, as expanding the list occurs on mouse-down, rather than mouse-up.

The spec for pointer cancellation is here:

https://www.w3.org/WAI/WCAG21/Understanding/pointer-cancellation.html
Here is a an example of an accessible combo box from wcag:
https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html

Note the difference in how their example #3 and how it works on mouse-down.

Click down on a combobox chevron without releasing the mouse button.

## 🤔 Expected Behavior

Clicking down your mouse should not cause list to expand and display the options.

## 😯 Current Behavior

Clicking down your mouse causes the list to expand and display the options.

## 💁 Possible Solution

Trigger for opening of list should be mouse up.

## 🔦 Context

Using it in Marketo MEUE application. It is causing accessibility issue.

## 💻 Code Sample
```
import React from "react";
import { Provider, defaultTheme, ComboBox, Item } from "@adobe/react-spectrum";


import "./styles.css";

export default function App() {
  return (
    <Provider theme={defaultTheme}>
      <ComboBox label="Favorite Animal">
        <Item key="red panda">Red Panda</Item>
        <Item key="cat">Cat</Item>
        <Item key="dog">Dog</Item>
        <Item key="aardvark">Aardvark</Item>
        <Item key="kangaroo">Kangaroo</Item>
        <Item key="snake">Snake</Item>
      </ComboBox>

    </Provider>
  );
}
```
## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | latest
| Browser          | Chrome 
| Operating System | macOS Big Sur

## 🧢 Your Company/Team

Adobe/Marketo - MEUE Team

## 🕷 Tracking Issue (optional)

https://jira.corp.adobe.com/browse/HG-153144

**Comment by devongovett** [comment]
This is intended and matches native behavior of menus, selects, and combo boxes which all open on mouse down.

Additionally, I do not believe this is a violation of the pointer cancelation success criteria, which states:

> For functionality that can be operated using a single pointer, at least one of the following is true:
>
> * No Down-Event: The down-event of the pointer is not used to execute any part of the function;
> * Abort or Undo: Completion of the function is on the up-event, and a mechanism is available to abort the function before completion or to undo the function after completion;
> * Up Reversal: The up-event reverses any outcome of the preceding down-event;
> * Essential: Completing the function on the down-event is essential.

In particular, the "Abort or Undo" case is met. Opening the menu does not perform any action that cannot be reversed trivially. Actual selection of options occurs on mouse up.
